src/wpsmon.c

```diff
@@ -450,7 +479,7 @@ void parse_wps_settings(const u_char *packet, struct pcap_pkthdr *header, char *
 				wps_parsed = parse_wps_parameters(packet, header->len, wps);
 				if(is_beacon || !get_ap_vendor(bssid)) set_ap_vendor(bssid);
 			}
-			if(!is_done(bssid, is_probe_resp && wps_parsed ? wps : 0) && (get_channel() == channel || source == PCAP_FILE))
+			if(!is_done(bssid, wps_parsed ? wps : 0) && (get_channel() == channel || source == PCAP_FILE))
```

Comment:
i added the is_probe_resp condition because checksum is always different with beacon vs probe_resp. with the condition removed, the checksum-related code will now clear complete flag whenever we receive a probe resp vs a beacon

Reply:
In regular mode, with the is_probe_resp condition and after AP was printed, delay to print and maybe not prints APs which have PBC on. I'll think of another way to solve this.

Reply:
is_probe_resp will be the third parameter for is_done()

Reply:
i think we have 3 cases which should trigger a reprint
- change of locked state
- change of PBC state
- general change of WPS state (like WPS info completely disappearing from some router models after a failed attack)

we could model all of these 3 cases as flags and get rid of the checksum.

Reply:
> change of locked state

New flag for it, done.

> change of PBC state

New flag for it, done.

> general change of WPS state (like WPS info completely disappearing from some router models after a failed attack)

New flag for it, done. But needs `-a`